src/sentry/api/endpoints/project_details.py

```diff
@@ -142,7 +157,14 @@ def validate(self, data):
         try:
             config_str = json.dumps(data)
             validate_sampling_configuration(config_str)
-            self.validate_uniform_sampling_rule(data.get("rules", []))
+
+            rules = data.get("rules", [])
+            self.validate_uniform_sampling_rule(rules=rules)
+            self.authorize_dynamic_sampling(
```

Comment:
Should authorization come before validation?

Reply:
I agree that it should come before but to understand the data we need to first validate it (e.g., checking if the number of **valid** rules is > 1). How would you say to a user, hey you can't do that if you don't even know what "that" is?